File: android/app/src/main/java/com/coinswallet/coins/DogeCoin.java
// BitcoinCoin.java
package com.coinswallet.coins;

import com.coinswallet.CoinFactory;
import com.coinswallet.Utils;
import wallet.core.jni.CoinType;
import wallet.core.jni.HDWallet;


public class DogeCoin extends CoinFactory.Coin {
    private final HDWallet wallet;

    public DogeCoin(String mnemonic) {
        super(mnemonic);
        this.wallet = super.wallet;
    }

    @Override
    public String getNewAddress(Boolean isTestNet) {
        // Derive a new private key for each call
        return CoinType.DOGECOIN.deriveAddress(wallet.getKeyForCoin(CoinType.DOGECOIN));
    }

    @Override
    public String getPrivateKey(Boolean isTestNet) {
        byte[] privateKeyBytes = wallet.getKeyForCoin(CoinType.DOGECOIN).data();
        byte[] prefix = new byte[]{(byte) 0x9e};
        return Utils.convertPrivateKeytoWIF(privateKeyBytes,false,prefix,prefix);
    }

    @Override
    public String signTransaction(String rawData) {
        // You need to implement this method according to the specific requirements of
        // Bitcoin transactions.
        return null;
    }
}

File: ios/DogeCoin.swift
//
//  BitcoinCoin.swift
//  coinswallet
//
//  Created by Tal Grynbaum on 30/06/2023.
//


import Foundation
import WalletCore

class DogeCoin: CoinFactory.Coin {
  private var addressIndex: Int = 0
  
  override init(mnemonic: String) {
    super.init(mnemonic: mnemonic)
  }
  
  override func getNewAddress(isTestNet:Bool) -> String {
    let address = CoinType.dogecoin.deriveAddress(privateKey: wallet.getKeyForCoin(coin: .dogecoin))
    return address
  }
  
  override func getPrivateKey(isTestNet:Bool) -> String {
    let privateKeyBytes = wallet.getKeyForCoin(coin: .dogecoin).data
    return Utils.convertToWif(data: privateKeyBytes, isTestNet: false, prefix: [0x9e], testNetPrefix: [0x9e])
    
  }
  
  override func signTransaction(rawData: String) -> String {
    // Implement the logic to sign a Bitcoin transaction using the provided raw data
    return ""
  }
}
